src/Core/Style.js

```diff
@@ -456,8 +499,14 @@ class Style {
             this.text.opacity = properties['label-opacity'];
             this.text.size = properties['label-size'];
 
-            if (properties.icon) {
-                this.icon.source = properties.icon;
+            const icon = {
+                ...(properties.icon !== undefined && { source: properties.icon }),
+                ...(properties['icon-scale'] !== undefined && { size: properties['icon-scale'] }),
+                ...(properties['icon-opacity'] !== undefined && { opacity: properties['icon-opacity'] }),
+                ...(properties['icon-color'] !== undefined && { color: properties['icon-color'] }),
+            };
+            if (Object.keys(icon).length) {
+                this.icon = icon;
```

Comment:
could you modify `defineStyleProperty` to not create property if feature property doesn't exist

Reply:
I got the point and will try to do it in the following PR when i really try to work on Style